### ISSUE TITLE
YouTube device support message 

### ISSUE THREAD

**Issue by racafe** [jakiestfu/Youtube-TV#44]
It appears that the new version of YouTube is upgrading to a new version which is not going to be supported with this plugin, does anyone know how to solve this?

![youtube](https://cloud.githubusercontent.com/assets/1800897/7237124/1109789a-e75f-11e4-8f53-aff545f44789.jpg)

**Comment by HeathNaylor** [comment]
This is not accurate, I just configured this plugin today with no issues at all, can you post a gist of your code? Seems like something in your options is off.

**Comment by racafe** [comment]
@HeathNaylor I actually put the screenshot from the demo: http://jakiestfu.github.io/Youtube-TV/demos/default.html

Check it out, it works but it preppends to the list a message that is not going to be supported with the new version of YouTube API.

**Comment by HeathNaylor** [comment]
Interesting, mine is working in a production website right now, I specified a string for the playlist id.

**Comment by beweinreich** [comment]
I'm getting same thing on production website. 

Looks like YouTube decided to phase out v2 of their API today. See here: http://youtube-eng.blogspot.com/2014/03/committing-to-youtube-data-api-v3_4.html

**Comment by fnezrick** [comment]
yes im seeing this too...  it is even prepending on the demos on github:  http://jakiestfu.github.io/Youtube-TV/demos/default.html 

Does anyone have a fix for this?  or is it time to move to a new plugin?  Thanks!

**Comment by beweinreich** [comment]
There's no quick fix. The code would have to be updated to use the V3 API. 

I might have some time this weekend to submit a pull request, but if there are any brave JavaScript developers who want to tackle it first... Do it!

**Comment by bradleywest92** [comment]
Hi guys, did you produce a fix for this at all?

**Comment by HeathNaylor** [comment]
I would just like to point out that mine is still fine in production, I am using the playlist option.

**Comment by bradleywest92** [comment]
Is that without the other video coming first Heath? If so would love to see your one in action if you didn't mind?

**Comment by eanbowman** [comment]
I did a 10-second fix to our production website with the same issue by editing the iframe in-browser.

Removing all of the settings and just putting youtube.com/embed/<<insert your id here>> worked.

Not sure what to do about the settings other than update them to v3 format as well (if they're supported).

**Comment by mrstork** [comment]
Try this fork https://github.com/Giorgio003/Youtube-TV
It uses the v3 api so you will need to enter your google developer key, but solves the issue

**Comment by HeathNaylor** [comment]
We have too many forks. @jakiestfu can we get a commitment from you that your branch will be updated and considered best?

**Comment by Giorgio003** [comment]
Chiming in here, I did managed to tackle API V3 a few weeks ago which does require an API Key to use. However you can obtain one for free ([https://developers.google.com/youtube/v3/getting-started](https://developers.google.com/youtube/v3/getting-started)) which gives a generous 50 million token requests per day.  
Should be plenty for most projects however I look to optimize this further by removing some unused variables within Youtube-TV, and possibly introduce some caching to prevent some repeated token requests on previous data.  
Apart from my changes to documentation, responsiveness, additional 'light' theme, etc. There is still some other issues I wish to attend to before pushing; Cleaning up the code by optimizing and removing unused code (ie. `loadAds` function is never used as the ads integrated into the Youtube player).  
I was going to mention the visible key however referrers seem to be working again for browser applications key allowing you to limit by domain or IP preventing any external use.  
As @HeathNaylor mentioned too many forks can get confusing overtime, I can PR into @jakiestfu 's repo if he's interested, or if not I created a reference markdown while I was converting from APIv2 to APIv3 ([APIv2-3_Reference.md](https://github.com/Giorgio003/Youtube-TV/blob/master/APIv2-3_Reference.md) it might miss some last minute tweaks but it contains all the variable and url name changes). Either way I have enabled the issue tracker on my fork in case anyone comes across bugs that need to be squashed.

**Comment by stevebrush** [comment]
Thanks so much Giorgio003! Your work saved me many hours.

**Comment by bachors** [comment]
I've changed to v3 http://ibacor.com/download/demo/youtube-channels/
